src/Bicep.Core/Semantics/SemanticModel.cs

```diff
@@ -86,17 +94,36 @@ public IReadOnlyList<Diagnostic> GetSemanticDiagnostics()
 
             var typeValidationDiagnostics = TypeManager.GetAllDiagnostics();
             diagnosticWriter.WriteMultiple(typeValidationDiagnostics);
-
             diagnosticWriter.WriteMultiple(EmitLimitationInfo.Diagnostics);
 
             return diagnosticWriter.GetDiagnostics();
         }
 
+        /// <summary>
+        /// Gets all the analyzer diagnostics unsorted.
+        /// </summary>
+        /// <returns></returns>
+        public IReadOnlyList<IDiagnostic> GetAnalyzerDiagnostics(ConfigHelper? overrideConfig = default)
+        {
+            if (overrideConfig != default)
+            {
+                LinterAnalyzer.OverrideConfig(overrideConfig);
+            }
+
+            var diagnostics = LinterAnalyzer.Analyze(this, overrideConfig);
+            var diagnosticWriter = ToListDiagnosticWriter.Create();
+            diagnosticWriter.WriteMultiple(diagnostics);
+
+            return diagnosticWriter.GetDiagnostics();
+        }
+
         /// <summary>
         /// Gets all the diagnostics sorted by span position ascending. This includes lexer, parser, and semantic diagnostics.
         /// </summary>
-        public IEnumerable<Diagnostic> GetAllDiagnostics() => GetParseDiagnostics()
+        public IEnumerable<IDiagnostic> GetAllDiagnostics(ConfigHelper? overrideConfig = default) =>
```

Comment:
>ConfigHelper [](http://example.com/codeflow?start=58&length=12)

The ConfigHelper part here feels looks tacked on to me. Can we first class the config file as something that is present on the Compilation or SemanticModel? (With an ability to override default config discovery behavior when a compilation is being constructed via an API in tests.)